components/tidb_query/src/expr/builtin_json.rs

```diff
@@ -81,6 +81,20 @@ impl ScalarFunc {
         Ok(j.extract(&path_exprs).map(Cow::Owned))
     }
 
+    pub fn json_length<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let j = try_opt!(self.children[0].eval_json(ctx, row));
+        let parser = JsonFuncArgsParser::new(row);
+        let path_exprs: Vec<_> = match parser.get_path_exprs(ctx, &self.children[1..])? {
```

Comment:
I think when the second argument is not provided, the length of `self.children`  is 1, `self.children[1..]` will panic. 
so you need to add a test case like `let op = scalar_func_expr(ScalarFuncSig::JsonLengthSig, &[json]);`

Reply:
Thank you. I wrongly understand ```scalar_func_expr```.